src/InstagramClone.Api/Database/InstagramCloneDbContext.cs

```diff
@@ -0,0 +1,41 @@
+﻿using InstagramClone.Api.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace InstagramClone.Api.Database
+{
+    public class InstagramCloneDbContext : DbContext
+    {
+        public InstagramCloneDbContext(): base()
+        {
+
+        }
+
+       
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<User>().HasKey(u => u.Id);
+            modelBuilder.Entity<Post>().HasKey(p => p.Id);
+            modelBuilder.Entity<PostReaction>().HasKey(pr => pr.Id);
+
+            // composite Key
+            modelBuilder.Entity<UserFollower>().HasKey(userfollower => new {userfollower.FollowedUserId, userfollower.FollowerId});
+            modelBuilder.Entity<PostReaction>().HasKey(postreaction => new {postreaction.PostId, postreaction.AuthorId} );
```

Comment:
really think about it...  what you're trying to do here is to configure the primary key of your table.

Based on our past conversation, What is the primary key of PostReaction?

Reply:
Now I get it, I added postReaction because it has two FK, which userFollower table also has. But after taking a close look at it I can see that my postReaction already has it own PK and userFollower does not have a PK that is why we used the composite Key to set a primary.
I will make the changes if I am not wrong?

Reply:
no, actually you nailed it! this is the exact explanation for it. so yes, make the change